src/entities/UserContextObject.ts

```diff
@@ -0,0 +1,8 @@
+import { Customer } from "./Customer";
+
+export interface UserContextObject {
+  user: Customer | null;
```

Comment:
I personally prefer to write like this: "user?: Customer".

Reply:
Arguably more concise indeed. Removed " | null" and the `initialUserState` and passed `{} as UserContextObject` to initialize the context.